src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -633,7 +633,7 @@ namespace syrec {
             }
 
         } else {
-            codewords = encodeHuffman(tt);
+            codewords = encodeWithoutAdditionalLine ? encodeHuffman(tt) : encodeHuffman(tt, false);
```

Comment:
```suggestion
            codewords = encodeHuffman(tt,  encodeWithoutAdditionalLine);
```
Following the argumentation above, this probably has to be refactored anyway, but the above was a very clumsy way of writing that code.